src/azure-cli/azure/cli/command_modules/storage/operations/blob.py

```diff
@@ -728,47 +728,88 @@ def _delete_blob(blob_name):
+                                  '_shared_access_signature#generate_blob_sas')
+
+    account_name = client.account_name
+    user_delegation_key = None
+    account_key = None
     if as_user:
         user_delegation_key = client.get_user_delegation_key(
             get_datetime_from_string(start) if start else datetime.utcnow(), get_datetime_from_string(expiry))
-        sas_token = client.generate_blob_shared_access_signature(
-            container_name, blob_name, permission=permission, expiry=expiry, start=start, id=id, ip=ip,
-            protocol=protocol, cache_control=cache_control, content_disposition=content_disposition,
-            content_encoding=content_encoding, content_language=content_language, content_type=content_type,
-            user_delegation_key=user_delegation_key)
     else:
-        sas_token = client.generate_blob_shared_access_signature(
-            container_name, blob_name, permission=permission, expiry=expiry, start=start, id=id, ip=ip,
-            protocol=protocol, cache_control=cache_control, content_disposition=content_disposition,
-            content_encoding=content_encoding, content_language=content_language, content_type=content_type)
+        account_key = client.credential.account_key
+
+    blob_url = kwargs.pop('blob_url')
+    container_name = kwargs.pop('container_name')
+    blob_name = kwargs.pop('blob_name')
+    if blob_url:
+        t_blob_client = get_sdk(cmd.cli_ctx, ResourceType.DATA_STORAGE_BLOB, '_blob_client#BlobClient')
+        if as_user:
+            credential = client.credential._credential
+        else:
+            credential = client.credential.account_key
+        blob_client = t_blob_client.from_blob_url(blob_url=blob_url, credential=credential, snapshot=snapshot)
+        container_name = blob_client.container_name
+        blob_name = blob_client.blob_name
+    else:
+        blob_client = client.get_blob_client(container=container_name, blob=blob_name, snapshot=snapshot)
+        blob_url = blob_client.url
+
+    sas_token = t_generate_blob_sas(account_name=account_name, container_name=container_name, blob_name=blob_name,
+                                    snapshot=snapshot, account_key=account_key, user_delegation_key=user_delegation_key,
+                                    permission=permission, expiry=expiry, start=start, policy_id=id, ip=ip,
+                                    protocol=protocol, cache_control=cache_control,
+                                    content_disposition=content_disposition, content_encoding=content_encoding,
+                                    content_language=content_language, content_type=content_type, **kwargs)
+
     if full_uri:
-        return encode_url_path(client.make_blob_url(container_name, blob_name, protocol=protocol,
-                                                    sas_token=quote(sas_token, safe='&%()$=\',~')))
+        return encode_url_path(make_blob_url(url=blob_url, protocol=protocol,
```

Comment:
Why not use `blob_client.url` but manually implemented `make_blob_url`?

Reply:
It combines the sas_token with the url.